includes/data-stores/class-wc-admin-reports-customers-data-store.php

```diff
@@ -391,6 +391,34 @@ public function get_data( $query_args ) {
 		return $data;
 	}
 
+	/**
+	 * Gets the customer ID for given order, either from user_id, if the order was done
+	 * by a registered customer, or from the billing email in the provided WC_Order.
+	 *
+	 * @param WC_Order $order Order to get customer data from.
+	 * @return int|false The ID of the retrieved customer, or false when there is no such customer.
+	 */
+	public static function get_customer_id_from_order( $order ) {
+		$user_id = $order->get_user_id();
+		if ( $user_id ) {
+			return self::get_customer_id_by_user_id( $user_id );
+		}
+
+		$email = $order->get_billing_email( 'edit' );
```

Comment:
If we go with the above refactor, I think we can drop the rest of this function and return the following:

`return get_or_create_guest_customer_from_order( $order );`